src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -0,0 +1,457 @@
+                        display = "float.NegativeInfinity";
+                    else if (f == float.MaxValue)
+                        display = "float.MaxValue";
+                    else if (f == float.MinValue)
+                        display = "float.MinValue";
+                    else
+                    {
+                        if (display.IndexOf('.') == -1)
+                            display += ".0";
+                        display += "f";
+                    }
+                }
+                else if (arg is decimal)
+                {
+                    decimal d = (decimal)arg;
+                    if (d == decimal.MinValue)
+                        display = "decimal.MinValue";
+                    else if (d == decimal.MaxValue)
+                        display = "decimal.MaxValue";
+                    else
+                        display += "m";
+                }
+                else if (arg is long)
+                {
+                    long l = (long)arg;
+                    if (l == long.MinValue)
+                        display = "long.MinValue";
+                    else if (l == long.MinValue)
+                        display = "long.MaxValue";
+                    else
+                        display += "L";
+                }
+                else if (arg is ulong)
+                {
+                    ulong ul = (ulong)arg;
+                    if (ul == ulong.MinValue)
+                        display = "ulong.MinValue";
+                    else if (ul == ulong.MinValue)
+                        display = "ulong.MaxValue";
+                    else
+                        display += "UL";
+                }
+                else if (arg is string)
+                {
+                    var str = (string)arg;
+                    bool tooLong = str.Length > stringMax;
+                    int limit = stringMax - THREE_DOTS.Length;
+
+                    StringBuilder sb = new StringBuilder();
+                    sb.Append("\"");
+                    foreach (char c in (string)arg)
+                    {
+                        sb.Append(EscapeControlChar(c));
+                        if (tooLong && sb.Length > limit)
+                        {
+                            sb.Append(THREE_DOTS);
+                            break;
+                        }
+                    }
+                    sb.Append("\"");
+                    display = sb.ToString();
+                }
+                else if (arg is char)
+                {
+                    display = "\'" + EscapeControlChar((char)arg) + "\'";
+                }
+                else if (arg is int)
+                {
+                    int ival = (int)arg;
+                    if (ival == int.MaxValue)
+                        display = "int.MaxValue";
+                    else if (ival == int.MinValue)
+                        display = "int.MinValue";
+                }
```

Comment:
This is old code I didn't modify, just copied it from the extinct MethodHelper class. But you're probably right and this is as good a time as any to do it.